### ISSUE TITLE
[Feature] Enhanced rendering of blob outputs based on mime types

### ISSUE THREAD

**Issue by schottra** [lyft/flyte#526]
**Motivation: Why do you think this is important?**
Workflows sometimes generate output that is more complex than just a series of numbers/text. A common use case is to generate plots of data to help visualize the result of a model. These are usually created by setting the output type of a Workflow to `Blob` and filling in the `format` field to indicate the file type.
It would be useful and more efficient for users if Flyte Console supported rendering certain blobs natively based on the `format` field.
Examples:
* Common image formats could be loaded directly in the browser.
* HTML documents could be loaded in an iframe.

**Goal: What should the final outcome look like, ideally?**
Workflows which output a blob with a known format will have a link/button in the UI to expand. Clicking this will attempt to load the blob at the given URL and display it in the browser.

**Describe alternatives you've considered**
We have previously discussed creating [custom rendering components](https://github.com/lyft/flyte/issues/146) for various task types, which would allow us to natively generate plots and other types of output. This would provide ultimate flexibility in terms of user interaction. But it is prohibitively complicated and blurs the line of responsibility for Flyte. Additionally, many tools already exist and are commonly used to generate these visual outputs in recognizable formats. We only need to provide a way to host the output.

**Flyte component**
- [ ] Overall
- [ ] Flyte Setup and Installation scripts
- [ ] Flyte Documentation
- [ ] Flyte communication (slack/email etc)
- [ ] FlytePropeller
- [ ] FlyteIDL (Flyte specification language)
- [ ] Flytekit (Python SDK)
- [ ] FlyteAdmin (Control Plane service)
- [ ] FlytePlugins
- [ ] DataCatalog 
- [ ] FlyteStdlib (common libraries)
- [x] FlyteConsole (UI)
- [ ] Other

**[Optional] Propose: Link/Inline**
High-level pieces:
* Blob `format` fields specify a proper mime type instead of an arbitrary string which only has meaning to the user. We need to decide how much to force this, whether or not Flyte will attempt to detect and set mime types, etc. For now, it's probably okay to just document the behavior and leave it up to users to specify the correct mime type in order to enable native rendering in Console.
  * Alternatively, console could attempt to do mime-type detection. But that requires fetching the blob, so it's not ideal.
* Add support to console for visualizing a blob. This could be as simple as a modal overlay that attempts to load the provided content in whatever way is most appropriate for the mime type.
  * We may have to adjust how inputs/outputs are rendered at the WorkflowExecution level, since they are already shown in a modal and we don't want to show an *additional* modal on top of that.

**Comment by kumare3** [comment]
@wild-endeavor we need to ensure that the blob format is also added as a header to the objects stored in s3/gcs etc

**Comment by wild-endeavor** [comment]
It looks like the s3 head operation will return the contenttype that you set on upload. https://docs.aws.amazon.com/cli/latest/reference/s3api/head-object.html
Is this what you were thinking?  I think it accepts arbitrary strings.  My vote is to just store whatever format is in the Blobtype in that field.

**Comment by kumare3** [comment]
> It looks like the s3 head operation will return the contenttype that you set on upload. https://docs.aws.amazon.com/cli/latest/reference/s3api/head-object.html
> Is this what you were thinking? I think it accepts arbitrary strings. My vote is to just store whatever format is in the Blobtype in that field.

Yes, In flytekit we should provide a few constants (I guess they can just be http mime types that)
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types

It seems there is a mimetypes library that can also be used to guess the output type if not provided 
https://docs.python.org/3/library/mimetypes.html

**Comment by kumare3** [comment]
The format should be applied to
1. Type declaration
2. runtime metadata
3. header of the object

**Comment by wild-endeavor** [comment]
Not sure I understand what 1. 2. and 3. are.

I don't know how I feel about guessing file-types, nor translating file types to content-types.  do we really want to do this?
